test/m_f_c_coin.js

```diff
@@ -18,5 +84,350 @@ contract('MFC_Token', function(accounts) {
+        }).catch(function(e){
+            assert.equal(e, 'Error: VM Exception while processing transaction: revert', "something went wrong");
+        });
+    });
+});
+
+contract('MFC_Token tests burnAll', async function(accounts) {
+
+    /* Task 18 - Create test for MFC_coin burnAll() with Truffle */
+
+    /* Using Truffle, we check method for MFC_coin burnAll() and test 3 variants:
+       - burner have balance */
+    it("should transfer 10 tokens to second account and second account should burn all tokens", function() {
+        var user_balance_befor;
+        var owner = accounts[0];
+        var user = accounts[5];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.transfer(user, 10, {'from': owner});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            user_balance_befor = balance.valueOf();
+            return token_contract.burnAll({'from': user});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            return balance.valueOf();
+        }).then(function(user_balance_after){
+            assert.equal(user_balance_befor, 10, "something went wrong");
+            assert.equal(user_balance_after, 0, "something went wrong");
+        });
+    });
+
+    /* burner not have balance */
+    it("the second account does not have a record in the balance sheet and will try to burn all tokens", function() {
+        var user = accounts[6];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.burnAll({'from': user});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            return balance.valueOf();
+        }).then(function(balance) {
+            assert.equal(balance, 0, "something went wrong");
+        });
+    });
+
+    /* balance of burner = 0 */
+    it("should transfer 10 tokens to second account and second account should burn 10 tokens, then should bur all tokens", function() {
+        var owner = accounts[0];
+        var user = accounts[7];
+        var token_contract;
+        return token.deployed().then(function(instance) {
+            token_contract = instance;
+            return token_contract.transfer(user, 10, {'from': owner});
+        }).then(function() {
+            return token_contract.burn(10, {'from': user});
+        }).then(function() {
+            return token_contract.burnAll({'from': user});
+        }).then(function() {
+            return token_contract.balanceOf.call(user);
+        }).then(function(balance) {
+            return balance.valueOf();
+        }).then(function(balance) {
+            assert.equal(balance, 0, "something went wrong");
```

Comment:
You could write balance.valueOf(); in code without additional then function.
Also, in this case, no sense.